pjlib/include/pj/sock.h

```diff
@@ -1517,6 +1521,18 @@ PJ_DECL(pj_status_t) pj_sock_shutdown( pj_sock_t sockfd,
  *****************************************************************************
  */
 
+/**
+ * Deep clone the socket options.
+ *
+ * @param dst       Destination socket options.
+ * @param src       Source socket options.
+ *
+ * @return          PJ_SUCCESS on success, or PJ_ETOOBIG if
+ *                  the source's option length is too large.
+ */
+PJ_DECL(pj_status_t) pj_sockopt_params_clone(pj_sockopt_params *dst,
+                                             const pj_sockopt_params *src);
+
```

Comment:
Perhaps it is more flexible to clone using pool to handle socket options whose long values (e.g: byte array)?
Not sure if all existing apps have pool though.

Or alternatively, perhaps allow pool==NULL, which will use internal buffer, if sufficient.